File: Spring/Task 2 (Hierarchy)/sources/ice_cream/milky_ice_cream/ice_cream_cone/ice_cream_cone.cpp
//
// Created by rinsler on 27.03.18.
//

#include "ice_cream_cone.h"


IceCreamCone::IceCreamCone(
        std::string name,
        std::string manufacturer,
        Ingredient flavor,
        Ingredient cone,
        Ingredient glaze,
        Ingredient syrup
) :
        MilkyIceCream(name, manufacturer, flavor, glaze, syrup),
        cone(cone)
{

}

std::string IceCreamCone::getRecipe()
{
    return MilkyIceCream::getRecipe() +
           " Put it to the " + std::to_string(cone.weight) + "-gram " + cone.name + ".";
}

File: Spring/Task 2 (IceCream)/IceCreamSources/IceCreamImplementations/IceCreamCone.java
package IceCreamImplementations;

import com.sun.istack.internal.Nullable;

public class IceCreamCone extends MilkyIceCream {
    protected Ingredient cone;

    public IceCreamCone(String name, String manufacturer, Ingredient flavor, Ingredient cone, @Nullable Ingredient glaze, @Nullable Ingredient syrup) {
        super(name, manufacturer, flavor, glaze, syrup);
        this.cone = cone;
    }

    @Override
    public String getRecipe() {
        return super.getRecipe() +
                " Put it to the " + cone.getWeight() + "-gram " + cone.getName() + ".";
    }
}
